test/unittests/testutil.cc

```diff
@@ -245,7 +247,55 @@ MockCatalog* MockCatalog::AttachFreely(const std::string  &root_path,
   return new_catalog;
 }
 
-void MockCatalog::RegisterChild(MockCatalog *child) {
+void MockCatalog::RemoveChild(MockCatalog *child) {
```

Comment:
Ah sorry, looks like `AddChild` / `RemoveChild` is different from `RegisterNestedCatalog`.  Actually, what is the difference?